pkg/daemon/controller/controller.go

```diff
@@ -331,6 +338,22 @@ func (sc *ServiceController) reconcileService(service winsvc.Service, expected s
 	return sc.EnsureServiceState(service, svc.Running)
 }
 
+// ensureRecoveryActions sets the recovery actions for the service if they are configured incorrectly
+func (sc *ServiceController) ensureRecoveryActions(service winsvc.Service) error {
+	actions, err := service.RecoveryActions()
+	if err != nil {
+		return errors.Wrap(err, "error getting current recovery actions")
+	}
+
+	// Ensure the recovery action for the service is to restart 10 seconds after a crash.
+	expectedAction := mgr.RecoveryAction{Type: mgr.ServiceRestart, Delay: 10 * time.Second}
```

Comment:
Consider having a constant for recovery delay

Reply:
What happens if a service is constantly crashing and it keeps getting restarted? Is there a an upper limit for retries that we can set? Ideally we should retry N number of times and then WICD should generate an event.

Reply:
We can specify a sequence of actions to occur within a given time period, for example
Crash 1: Restart service after 1 min
Crash 2: Restart service after 2 min 
Crash 3: Restart service after 5 min
Crash 4+: Do nothing

And then 30 min after the first failure, reset the crash counter.

Generating a kubernetes event would have to be separate from this, as these counts are all managed within the Windows Service Manager